src/content/reference/react/cacheSignal.md

```diff
@@ -0,0 +1,121 @@
+---
+title: cacheSignal
+---
+
+<Canary>
+
+**The `cacheSignal()` API is currently only available in React’s Canary and Experimental channels.** 
+
+[Learn more about React’s release channels here.](/community/versioning-policy#all-release-channels)
+
+</Canary>
+
+<RSC>
+
+`cacheSignal` is only for use with [React Server Components](/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023#react-server-components).
+
+</RSC>
+
+<Intro>
+
+`cacheSignal()` returns an [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) which aborts when the cache lifetime is done based on the same execution scope as a [`cache()`](/reference/react/cache)ed function 
+
+```js
+const signal = cacheSignal();
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `cacheSignal` {/*cachesignal*/}
+
+Call `cacheSignal` to get an `AbortSignal`.
+
+```js {3,7}
+import {cacheSignal} from 'react';
+async function Component() {
+  await fetch(url, { signal: cacheSignal() });
+}
+```
+
+When React has finished rendering, the `AbortSignal` will be aborted. This allows you to cancel any in-flight work that is no longer needed.
+Rendering is considered finished when:
+- React has successfully completed rendering 
+- the render was aborted
+- the render has failed
+
+#### Parameters {/*parameters*/}
+
+This function does not accept any parameters.
+
+#### Returns {/*returns*/}
+
+`cacheSignal` returns an `AbortSignal` if called during rendering. Otherwise `cacheSignal()` returns `null`.
+
+#### Caveats {/*caveats*/}
+
+- `cacheSignal` is for use in [React Server Components](/reference/rsc/server-components) only. In Client Components, it will always return `null`.
```

Comment:
```suggestion
- `cacheSignal` is currently for use in [React Server Components](/reference/rsc/server-components) only. In Client Components, it will always return `null`. In the future it will also be used for Client Component when a client cache refreshes or invalidates. You should not assume it'll always be null on the client.
```